src/Model/Scope/AbstractCondition.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model\Scope;
+
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Exception;
+use atk4\data\Model;
+
+/**
+ * @property CompoundCondition $owner
+ */
+abstract class AbstractCondition
+{
+    use InitializerTrait {
+        init as _init;
+    }
+    use TrackableTrait;
+
+    /**
+     * Method is executed when the scope is added to parent scope using Scope::add
+     * $this->owner is the Scope object.
+     */
+    public function init(): void
+    {
+        if (!$this->owner instanceof self) {
+            throw new Exception('Scope can only be added as element to scope');
+        }
+
+        $this->_init();
+
+        $this->onChangeModel();
+    }
+
+    abstract public function onChangeModel();
+
+    /**
+     * Temporarily assign a model to the condition.
+     *
+     * @return static
+     */
+    final public function on(Model $model)
```

Comment:
does this method make any sense for anything other than `toWords`/`toArray`?

Reply:
Yes mostly for those 2 methods

Reply:
so is there anything against making `on` non-public and allowing to pass model directly to `toWords()`? (like also possibly for the future extention)

Reply:
If it's not useful to be used public, then make private and then we will see if anyone would like to use it as public or not. It's better than making all public and afterwards set it to private and break BC for developers.

Reply:
It is necessary for public at the moment to associate a condition object with a model.
Otherwise we have to pass a model object as optional argument to `Condition::toWords` method and use it only if `Condition::getModel` is empty

Reply:
> 
> 
> It is necessary for public at the moment to associate a condition object with a model.
> Otherwise we have to pass a model object as optional argument to `Condition::toWords` method and use it only if `Condition::getModel` is empty

I think it is cleaner to allow to pass optional Model to toWords. We should not need a clone then at all.

Reply:
Changed now, resolved?